### PR TITLE
Use <input type="time">

### PR DESCRIPTION
**Pull request by bershanskiy**
This is a draft because of two defects:
 - width is too small to fit the whole row
 - popup background can be light

image

image

### CODE REVIEW

**Comment by bershanskiy** [comment]
@Gusted This is ready for review now.

**Review by Gusted — APPROVED** [review]
LGTM code-wise. CSS/UI wise is up to @alexanderby :)

**Comment by Gusted** [comment]
Btw - might be interesting to see if it properly works on Android(firefox).

**Comment by bershanskiy** [comment]
I revived this PR and I think it is ready to be merged. I like `<input type="time">` over the current form since it is actually aware of user's time format preferences like 24-hour format vs. 12-hour AM/PM format. Also, input validation is much more ergonomic and the scroller just seems nicer.

File: src/ui/controls/textbox/style.less [hunk 1]
```diff
@@ -27,3 +27,9 @@
         color: @color-input-fore-placeholder;
     }
 }
+
+input[type="time"]::-webkit-calendar-picker-indicator{
```
Comment (Gusted) [inline src/ui/controls/textbox/style.less:31]:
```suggestion
input[type="time"]::-webkit-calendar-picker-indicator {
```

File: src/ui/controls/time-range-picker/index.tsx [hunk 2]
```diff
@@ -69,7 +57,8 @@ export default function TimeRangePicker(props: TimePickerProps) {
             />
             <TextBox
                 class="time-range-picker__input time-range-picker__input--end"
-                placeholder={toLocaleTime('9:00')}
+                type="time"
+                placeholder="9:00"
```
Comment (Gusted) [inline src/ui/controls/time-range-picker/index.tsx:61]:
```suggestion
                placeholder="09:00"
```

**Review by Gusted — APPROVED** [review]

**Comment by bershanskiy** [comment]
On Firefox for Android, `<input type="time">` actually looks even better. You can [test it on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time).

image

**Pull request merged**